controllers/topics/getLatestPost.js

```diff
@@ -0,0 +1,58 @@
+const {QueryTypes} = require('sequelize');
+const {sequelize} = require('../../databases/models');
+
+module.exports = async (req, res) => {
+  const {name} = req.query;
+  try {
+    let topics = await sequelize.query(
+      `SELECT topic_id FROM topics
+      WHERE name = :name`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          name
+        }
+      }
+    );
+    let topicIds = topics.map((topic) => topic.topic_id);
+
+    if (topicIds.length === 0) {
+      res.status(400).json({
+        code: 400,
+        message: 'Topic not found'
+      });
+    }
+
+    let post_topics = await sequelize.query(
+      `SELECT
+        pt.topic_id,
+        p.*
+      FROM posts p 
+        INNER JOIN post_topics pt 
+        ON p.post_id = pt.post_id
+      WHERE 
+        pt.topic_id IN (:topicIds)
+        AND p.duration >= now()
+      ORDER BY created_at DESC
+      LIMIT 1`,
+      {
+        type: QueryTypes.SELECT,
+        replacements: {
+          topicIds
+        }
+      }
+    );
+
+    res.status(200).json({
+      status: 'success',
+      code: 200,
+      data: post_topics
+    });
+  } catch (error) {
+    console.log(error);
+    res.status(500).json({
+      code: 500,
+      message: 'Internal server error'
+    });
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code seems to be missing error handling for the case when no posts are found for a given topic. If `post_topics` is empty, it would be better to return a 404 status with an appropriate message instead of returning a success status with an empty data array.

```diff
+    if (post_topics.length === 0) {
+      res.status(404).json({
+        code: 404,
+        message: 'No posts found for this topic'
+      });
+      return;
+    }

     res.status(200).json({
       status: 'success',
       code: 200,
       data: post_topics
     });
```

Also, it's not a good practice to log the entire error object directly to the console as it may contain sensitive information. Instead, you should consider using a logging library that can sanitize the error messages or create custom error messages.

```diff
-    console.log(error);
+    console.error(`Error occurred while fetching latest post: ${error.message}`);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->